### PR TITLE
Adding csharpier to format-all

### PR DESCRIPTION
**Pull request by passeride**
Adding this [csharpier](https://github.com/belav/csharpier) as the default formatter for C# files.

### CODE REVIEW

File: format-all.el [hunk 1]
```diff
@@ -830,6 +830,12 @@ Consult the existing formatters for examples of BODY."
   (:features)
   (:format (format-all--buffer-easy executable "fmt")))
 
+(define-format-all-formatter csharpier
+  (:executable "dotnet-csharpier")
+  (:install "dotnet install -g csharpier")
+  (:languages "C#")
+  (:features)
+  (:format (format-all--buffer-easy executable "--write-stdout")))
```
Comment (lassik) [inline format-all.el:838]: * Please keep the formatter definitions in alphabetical order.
* Please ensure there is a blank line above and below the formatter definition.

Comment (passeride) [reply format-all.el]: :heavy_check_mark:  Moved and formatted formatter definition in formatter package

File: format-all.el [hunk 2]
```diff
@@ -29,7 +29,7 @@
 ;; - Bazel Starlark (buildifier)
 ;; - BibTeX (Emacs)
 ;; - C/C++/Objective-C (clang-format, astyle)
-;; - C# (clang-format, astyle)
+;; - C# (clang-format, astyle, csharpier)
```
[the remark this answers is not in the document]
Comment (passeride) [reply format-all.el:32]: :thumbsup:

**Comment by lassik** [comment]
Thanks. LTGM

**Comment by belav** [comment]
I just released csharpier 1.0 which has some breaking changes. 

- executable is now just `csharpier`
- you now need to explicitly pass format, `dotnet csharpier format .` vs the old way of `dotnet csharpier .`

I'm guessing the changes would look something like what is below, but I don't know if there is a way to account for versions. This should only be used for >= 1.0

```
(define-format-all-formatter csharpier
  (:executable "csharpier")
  (:install "dotnet install -g csharpier")
  (:languages "C#")
  (:features)
  (:format (format-all--buffer-easy executable "format --write-stdout")))
```

**Comment by knu** [comment]
Does `dotnet install` have an option to specify version constraints?

**Comment by belav** [comment]
> Does `dotnet install` have an option to specify version constraints?

Yeah - if you wanted to install the last pre-1.0.0 version it would be

```
dotnet tool install csharpier -g --version 0.30.6
```

` --allow-downgrade` would be required if someone has already installed `1.0.0`

**Pull request merged**